missing js files

following the steps here:
https://github.com/jupyter/jupyter_notebook#jupyter-notebook
Dev quickstart:

I see:
```
(jup)carl@twist:~/src/jupyter_notebook$ jupyter notebook
[I 22:45:35.585 NotebookApp] Serving notebooks from local directory: /home/carl/src/jupyter_notebook
[I 22:45:35.585 NotebookApp] 0 active kernels 
[I 22:45:35.585 NotebookApp] The IPython Notebook is running at: http://localhost:8888/
[I 22:45:35.585 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).

(process:23904): GLib-CRITICAL **: g_slice_set_config: assertion 'sys_page_size == 0' failed
[E 22:45:35.848 NotebookApp] Could not open static file ''
[W 22:45:36.028 NotebookApp] 404 GET /static/components/jquery-ui/themes/smoothness/jquery-ui.min.css (::1) 14.14ms referer=http://localhost:8888/tree
[W 22:45:36.029 NotebookApp] 404 GET /static/style/style.min.css (::1) 0.66ms referer=http://localhost:8888/tree
[W 22:45:36.030 NotebookApp] 404 GET /static/components/es6-promise/promise.min.js (::1) 0.78ms referer=http://localhost:8888/tree
[W 22:45:36.032 NotebookApp] 404 GET /static/components/requirejs/require.js (::1) 0.93ms referer=http://localhost:8888/tree
```

trying setup.py:

```
(jup)carl@twist:~/src/jupyter_notebook$ python setup.py install
running install
running build
running build_py
running jsversion
running css
running js
npm unavailable
Failed to run bower: [Errno 2] No such file or directory
You can install js dependencies with `npm install`
rebuilding css and sourcemaps failed (not a problem)
[Errno 2] No such file or directory
checking package data
Traceback (most recent call last):
  File "setup.py", line 187, in <module>
    main()
  File "setup.py", line 184, in main
    setup(**setup_args)
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/python2.7/distutils/command/install.py", line 601, in run
    self.run_command('build')
  File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/python2.7/distutils/command/build.py", line 128, in run
    self.run_command(cmd_name)
  File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/carl/src/jupyter_notebook/setupbase.py", line 450, in run
    command.run(self)
  File "/home/carl/src/jupyter_notebook/setupbase.py", line 193, in run
    check_package_data(self.package_data)
  File "/home/carl/src/jupyter_notebook/setupbase.py", line 183, in check_package_data
    assert os.path.exists(path), "Missing package data: %s" % path
AssertionError: Missing package data: jupyter_notebook/static/components/backbone/backbone-min.js
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -17,7 +17,7 @@ before_install:
     - 'if [[ $GROUP == js* ]]; then npm install -g casperjs; fi'
     - git clone --quiet --depth 1 https://github.com/minrk/travis-wheels travis-wheels
 install:
-    - pip install -f travis-wheels/wheelhouse -r requirements.txt -e file://$PWD#egg=jupyter_notebook[test] coveralls
+    - pip install -f travis-wheels/wheelhouse -r requirements.txt file://$PWD#egg=jupyter_notebook[test] coveralls
 script:
     - 'if [[ $GROUP == js* ]]; then python -m jupyter_notebook.jstest $GROUP; fi'
     - 'if [[ $GROUP == python ]]; then nosetests --with-coverage --cover-package=jupyter_notebook jupyter_notebook; fi'

--- a/setupbase.py
+++ b/setupbase.py
@@ -194,6 +194,13 @@ def check_package_data_first(command):
             command.run(self)
     return DecoratedCommand
 
+def update_package_data(distribution):
+    """update package_data to catch changes during setup"""
+    build_py = distribution.get_command_obj('build_py')
+    distribution.package_data = find_package_data()
+    # re-init build_py options which load package_data
+    build_py.finalize_options()
+
 #---------------------------------------------------------------------------
 # Notebook related
 #---------------------------------------------------------------------------
@@ -276,7 +283,7 @@ def mtime(path):
     """shorthand for mtime"""
     return os.stat(path).st_mtime
 
-py3compat_ns = {}
+
 
 class Bower(Command):
     description = "fetch static client-side components with bower"
@@ -334,7 +341,7 @@ class Bower(Command):
             raise
         os.utime(self.bower_dir, None)
         # update package data in case this created new files
-        self.distribution.package_data = find_package_data()
+        update_package_data(self.distribution)
 
 
 class CompileCSS(Command):
@@ -365,7 +372,7 @@ class CompileCSS(Command):
             print("You can install js dependencies with `npm install`", file=sys.stderr)
             raise
         # update package data in case this created new files
-        self.distribution.package_data = find_package_data()
+        update_package_data(self.distribution)
 
 
 class JavascriptVersion(Command):

